library/Frontend/src/Juvix/Frontend/Parser.hs

```diff
@@ -449,6 +451,64 @@ nameParser =
   P.try (P.skipSome (P.char J.hash) *> fmap Types.Implicit prefixSymbol)
     <|> Types.Concrete <$> prefixSymbol
 
+--------------------------------------------------
+-- Effect handler parser
+--------------------------------------------------
+
+handlerParser :: Parser Types.Handler
+handlerParser = do
+  reserved "handler"
+  name <- prefixSymbolSN
+  reserved "where"
+  ops <- P.sepEndBy opParser (skipLiner J.comma)
+  ret <- retParser
+  pure $ Types.Hand name ops ret
+
+opParser :: Parser Types.Operation
+opParser =
+  Types.Op
+    <$> functionModStartReserved
+      "op"
+      ( \name args -> do
+          guard <- guard expression
+          pure (Types.Like name args guard)
+      )
+
+retParser :: Parser Types.Operation
+retParser =
+  Types.Op <$> do
+    reserved "return"
+    args <- P.many argSN
+    guard <- guard expression
+    pure (Types.Like "return" args guard)
```

Comment:
Can this be turned into an `opParser`?

Reply:
Oooh, I thought I had refactored this to be an `opParser`, thanks.